### ISSUE TITLE
Add details for Remote Tracking Branches when cloning  a repository with multiple branches

### ISSUE THREAD

**Issue by ParkinT** [ParkinT/mastering_git#2]
Due to changes in Git 2.x the behavior with respect to remote-tracking branches (and 'fetch' or 'pull') is different.  Further complicating matters, all the top search results point to StackOverflow pages that were written prior to this change.

Now, a `git clone` command will **only** bring in the master branch.  In order to properly setup the remote-tracking-branches (when the cloned repository has other branches) you need to follow the advice in [this StackOverflow topic](http://stackoverflow.com/questions/10312521/how-to-fetch-all-git-branches) (about halfway down the page);
- Use this bash command to create local branches for all remote-tracking ones

```
for b in `git branch -r | grep -v -- '->'`; do git branch --track ${b##origin/} $b; done
```
- Use `'git fetch --all` or `git pull --all`

This has worked for me reliably many times.

> This needs to be properly described and included as part of the Addendum here.

**Comment by techmag** [comment]
Appreciate the update -- thank you. 

git is far from the only project that suffers from the "old links are generally better" policy of most search engines. You should see the mess this creates learning Play for example. I wonder if a more elegant solution to the problem can be found and applied globally? 

PS: I took many git courses and your rose to the top by a landslide...

**Comment by ParkinT** [comment]
Thanks, @techmag for the comment.  Primarily, I wanted to capture this for myself (all too often an Internet search results in a hit on something **I** wrote months/years ago!).
But, hopefully this will be useful for others in _that_ predicament.

> PS: I took many git courses and your rose to the top by a landslide...

I greatly appreciate the comment.  Please share this sentiment on social media!

**Comment by ParkinT** [comment]
Closed in [this commit](1f25cbed5a255aaeca2a68fb6e0e4e8c5075be3e)
